pkg/controller/controller.go

```diff
@@ -198,6 +199,10 @@ func (c *RolloutController) reconcile(ctx context.Context) error {
 	groups := groupStatefulSetsByLabel(sets, RolloutGroupLabel)
 	var reconcileErrs error
 	for groupName, groupSets := range groups {
+		if err := c.reconcileStatefulSetsGroupReplicas(ctx, groupName, groupSets); err != nil {
```

Comment:
I don't really like the way this functionality integrates with the existing reconcile loop, it feels "bolted on" to me. However, I did it like this to minimize changes to the existing logic and make it easier to develop. WDYT?

Reply:
Makes sense, thank you. I'll refactor a bit to move the replica reconciliation around.